backend/app/utils.py

```
@@ -3108,6 +3166,21 @@ def _find_map_json(directory):
+        # Skip metadata and certain system files
+        if "Metadata" in json_file.parts or json_file.name in ["GISProject.json", "Index.json"]:
+            continue
+
+        try:
+            with open(json_file, 'r', encoding='utf-8') as f:
+                content = f.read(2000)
+                # Look for map-like JSON with layers array
+                if '"type":"CIMMap"' in content and '"layers":[' in content:
+                    return json_file
+        except Exception:
+            continue
```

Ruff (0.15.0)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)